src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -549,7 +602,7 @@ private void addPathNodeToDebug(final Node node)
         debugNodesPath.add(node);
     }
 
-    private void walkCurrentNode(@NotNull final Node currentNode)
+    private void walkCurrentNode(@NotNull final Node currentNode, final boolean isInRestrictedArea)
```

Comment:
shouldn't we just make this a field then? instead of passing this around so much?

Reply:
like it's connected to all next nodes/poses from current node, not sure if feasible

Reply:
But it's also connected to one specific task right? so just making it a field should be fine

Reply:
It's connected to specific node, not task

Reply:
ahh, okay, then it's resolved

Reply:
Yeah this is fine

Reply:
However now that i see your use case, would you please put this in a seperate object? Something like `CurrentPathCandidateProperties` or `PathContext`. I think there could be uses for this kind of information in other places as well.